functional-tests/appwrapper_reconciler_test.go

```diff
@@ -0,0 +1,250 @@
+package functional_tests
+
+import (
+	"context"
+	"go/build"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	gstruct "github.com/onsi/gomega/gstruct"
+	machinev1beta1 "github.com/openshift/api/machine/v1beta1"
+	. "github.com/project-codeflare/codeflare-common/support"
+	"github.com/project-codeflare/instascale/controllers"
+	"github.com/project-codeflare/instascale/pkg/config"
+	mcadv1beta1 "github.com/project-codeflare/multi-cluster-app-dispatcher/pkg/apis/controller/v1beta1"
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	log "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var (
+	cfg       *rest.Config
+	k8sClient client.Client // You'll be using this client in your tests.
+	testEnv   *envtest.Environment
+	err       error
+)
```

Comment:
Ideally all these variables should be deleted. Everything should be passed within test function itself.